pkg/tsdb/azuremonitor/loganalytics/azure-response-table-frame_test.go

```diff
@@ -193,7 +194,418 @@ func TestLogTableToFrame(t *testing.T) {
 	}
 }
 
-func loadLogAnalyticsTestFileWithNumber(t *testing.T, name string) AzureLogAnalyticsResponse {
+func marshalJSONData(v interface{}) *json.RawMessage {
+	marshalled, err := json.Marshal(v)
+	if err != nil {
+		panic(fmt.Errorf("error marshalling data: %s", err))
+	}
+
+	raw := json.RawMessage(marshalled)
+	return &raw
+}
+
+func TestTraceTableToFrame(t *testing.T) {
+	tests := []struct {
+		name          string
+		testFile      string
+		expectedFrame func() *data.Frame
+		resultFormat  dataquery.ResultFormat
+	}{
+		{
+			name:     "multi trace",
+			testFile: "traces/1-traces-multiple-table.json",
+			expectedFrame: func() *data.Frame {
```

Comment:
This looks like a good use case for `CheckGoldenJSONResponse`. Here is an example of a helper function. https://github.com/grafana/github-datasource/blob/main/pkg/testutil/frames.go#L21

Reply:
I'll have a look at this 👀

Reply:
Thank you @bossinc for the great recommendation! That's a bunch of redundant code removed 😊